app/models/event.rb

```diff
@@ -880,6 +880,41 @@ def self.loop_through_events(options)
     Rails.logger.info("#{label}: task completed")
   end
 
+  def self.loop_through_gbif_events(options)
+    size = (options[:size] || 1_000).to_i
+    cursor = options[:cursor] || []
+    filter = options[:filter] || {}
+    label = options[:label] || ""
+    job_name = options[:job_name] || ""
+    query = options[:query].presence
+    delete_count = 0
+    max_delete_count = options[:max_delete_count]
+
+    response = Event.query(query, filter.merge(page: { size: 1, cursor: [] }))
+
+    if response.size.positive?
```

Comment:
we should verify `if response.present? && response.respond_to?(:size) && response.size.positive?` response present, incase of an error